new-integration-tests/src/test/java/oracle/weblogic/kubernetes/assertions/TestAssertions.java

```diff
@@ -25,57 +28,84 @@
 
   /**
    * Check if operator Rest service is running.
+   *
    * @param namespace the operator rest service exists
    * @return true if rest service is running otherwise false
    */
-  public static boolean operatorRestServiceRunning(String namespace) {
-    return Operator.isRestServiceCreated(namespace);
+  public static Callable<Boolean> operatorRestServiceRunning(String namespace) throws ApiException {
+    return () -> {
+      return Operator.isExternalRestServiceCreated(namespace);
+    };
   }
 
   /**
-   * Check if a WebLogic custom resource domain exists in specified namespace and all its pods are running.
+   * Check if a WebLogic custom resource domain object exists in specified
+   * namespace.
+   *
    * @param domainUID ID of the domain
-   * @param namespace in which the domain custom resource exists
-   * @return true if domain exists and pods running otherwise false
+   * @param namespace in which the domain custom resource object exists
+   * @return true if domain object exists
    */
   public static Callable<Boolean> domainExists(String domainUID, String namespace) {
     return Domain.exists(domainUID, namespace);
   }
 
+  /**
+   * Check if a Kubernetes pod exists in any state.
+   *
+   * @param podName name of the pod to check for
+   * @param domainUID WebLogic domain uid in which the pod belongs
+   * @param namespace in which the pod exists
+   * @return true if the pod exists in the namespace otherwise false
+   */
+  public static Callable<Boolean> podExists(String podName, String domainUID, String namespace) throws ApiException {
```

Comment:
Let's discuss in the meeting.  It seems that these assertions are insufficient (I know that they model exactly what was in the previous tests).  Just checking that the pod exists, is ready, or is terminating leaves off lots of possible errors.